apps/freeablo/faworld/potion.cpp

```diff
@@ -5,33 +5,31 @@
 
 namespace FAWorld
 {
-    Potion::Potion(){};
-    Potion::~Potion(){};
-
+    //https:// wheybags.gitlab.io/jarulfs-guide/#potions-and-elixirs
+    //Used for the formulas
     void Potion::restoreHp(Player* player)
     {
-        double bonus = 1.0;
-        // bonus came from Jarulf's guide
+        std::string bonus = "1.0";
+
         switch (player->getClass())
         {
             case FAWorld::PlayerClass::warrior: {
-                bonus = 2.0;
+                bonus = "2.0";
                 break;
             }
             case FAWorld::PlayerClass::rogue: {
-                bonus = 1.5;
+                bonus = "1.5f";
```

Comment:
These should all be FixedPoint expressions, not strings, eg: `FixedPoint bonus = 1; ... bonus = FixedPoint("1.5");`